pkg/tests/apis/alerting/rules/compat/alertrule_test.go

```diff
@@ -0,0 +1,421 @@
+package compat
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/grafana/grafana/apps/alerting/rules/pkg/apis/alerting/v0alpha1"
+	apimodels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+	ngmodels "github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/tests/api/alerting"
+	"github.com/grafana/grafana/pkg/tests/apis/alerting/rules/common"
+	"github.com/grafana/grafana/pkg/util"
+	prom_model "github.com/prometheus/common/model"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestIntegrationAlertRuleCompatCreateViaK8s(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	ctx := context.Background()
+	helper := common.GetTestHelper(t)
+
+	k8sClient := common.NewAlertRuleClient(t, helper.Org1.Admin)
+
+	legacyClient := alerting.NewAlertingLegacyAPIClient(helper.GetListenerAddress(), "admin", "admin")
+
+	// Ensure the old provisioning API is enabled
+	allRules, status, _ := legacyClient.GetAllRulesWithStatus(t)
+	require.Equal(t, 200, status)
+	require.NotNil(t, allRules)
+
+	// Create test folder first
+	common.CreateTestFolder(t, helper, "test-folder")
+
+	rule := ngmodels.RuleGen.With(
+		ngmodels.RuleMuts.WithUniqueUID(),
+		ngmodels.RuleMuts.WithUniqueTitle(),
+		ngmodels.RuleMuts.WithNamespaceUID("test-folder"),
+		ngmodels.RuleMuts.WithGroupName("test-group"),
+		ngmodels.RuleMuts.WithIntervalMatching(time.Duration(10)*time.Second),
+	).Generate()
+
+	alertRule := &v0alpha1.AlertRule{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "default",
+			Annotations: map[string]string{
+				"grafana.app/folder": "test-folder",
+				// use provenance api to allow use of the provisioning api
+				"grafana.app/provenance": string(ngmodels.ProvenanceAPI),
```

Comment:
we can use app extensions like SetProvenanceStatus etc so we can avoid sticking to annotation literals in our code.